charts/jaeger/templates/es-configmap.yaml

```diff
@@ -0,0 +1,62 @@
+{{ if (and (not .Values.userconfig) .Values.provisionDataStore.elasticsearch) }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-config
+  namespace: {{ include "jaeger.namespace" . }}
+  labels:
+    {{- include "jaeger.labels" . | nindent 4 }}
+data:
+  user-config.yaml: |
+    service:
+      extensions: [jaeger_storage, jaeger_query, healthcheckv2]
+      pipelines:
+        traces:
+          receivers: [otlp]
+          processors: [batch]
+          exporters: [jaeger_storage_exporter]
+
+    extensions:
+      healthcheckv2:
+        use_v2: true
+        http:
+            endpoint: 0.0.0.0:13133
+
+      jaeger_query:
+        storage:
+          traces: primary_store
+          traces_archive: archive_store
+
+
+      jaeger_storage:
+        backends:
+          primary_store:
+            elasticsearch:
```

Comment:
I don't mind merging like this as a milestone, but I think this file should not be specific to ES only, it can use the flags indicating which storage type the user selected and creating the corresponding configs for that storage. Otherwise we would have to duplicate this overall config for each storage type.